lib/tool_belt/koji_builder.rb

```diff
@@ -0,0 +1,163 @@
+require File.join(File.dirname(__FILE__), 'systools')
+
+module ToolBelt
+  class KojiBuilder
+
+    CLIENT_OSES = %w(rhel5 rhel6 rhel7 fedora22 fedora23)
+    SERVER_OSES = %w(rhel6 rhel7)
+    KOJI_COMMAND = 'koji -c ~/.koji/katello-config'
+
+    EXTERNAL_REPOS = {
+      'rhel5' => ['rhel-5.9-server', 'epel-5'],
+      'rhel6' => ['rhel-6.6-server', 'rhel-6.6-server-optional', 'epel-6'],
+      'rhel7' => ['rhel-7.0-server', 'rhel-7.0-server-optional', 'epel-7']
+    }
+
+    VERSIONS = {
+      'nightly' => 'nightly',
+      '2.0' => '1.6',
+      '2.1' => '1.7',
+      '2.2' => '1.8',
+      '2.3' => '1.9',
+      '2.4' => '1.10',
+      '3.0' => '1.11'
+    }
+
+    attr_reader :version, :systools
+
+    def initialize(args = {})
+      @katello_version = args.fetch(:katello_version, 'nightly') || 'nightly'
+      @systools = args.fetch(:systools) || SysTools.new
+      @foreman_version = VERSIONS[@katello_version]
+    end
+
+    def build_katello_tags(os)
+      operating_systems = os.nil? ? CLIENT_OSES : os
```

Comment:
Switched this to all OSes essentially for building non-server OS tags such as katello-nightly-fedora23